### ISSUE TITLE
Error: Cannot find module 'babel-eslint'

### ISSUE THREAD

**Issue by willin** [AtomLinter/linter-eslint#556]
```
Error: Cannot find module 'babel-eslint'
Error: Cannot find module 'babel-eslint'
    at Object.ModuleResolver.resolve (/Users/willin/.atom/packages/linter-eslint/node_modules/eslint/lib/util/module-resolver.js:77:19)
    at Object.load (/Users/willin/.atom/packages/linter-eslint/node_modules/eslint/lib/config/config-file.js:515:42)
    at loadConfig (/Users/willin/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:67:33)
    at new Config (/Users/willin/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:200:38)
    at CLIEngine.executeOnText (/Users/willin/.atom/packages/linter-eslint/node_modules/eslint/lib/cli-engine.js:655:28)
    at Object.cli.execute (/Users/willin/.atom/packages/linter-eslint/node_modules/eslint/lib/cli.js:180:36)
    at lintJob (/Users/willin/.atom/packages/linter-eslint/lib/worker.js:30:10)
    at /Users/willin/.atom/packages/linter-eslint/lib/worker.js:66:20
    at Emitter.emit (/Users/willin/.atom/packages/linter-eslint/node_modules/process-communication/node_modules/sb-event-kit/lib/emitter.js:70:19)
    at /Users/willin/.atom/packages/linter-eslint/node_modules/process-communication/node_modules/sb-communication/lib/main.js:22:23
```

```
tree node_modules
node_modules
├── babel-cli -> .npminstall/babel-cli/6.7.5/babel-cli
├── babel-eslint -> .npminstall/babel-eslint/6.0.3/babel-eslint
├── babel-plugin-proxy -> .npminstall/babel-plugin-proxy/1.0.6/babel-plugin-proxy
├── babel-plugin-transform-runtime -> .npminstall/babel-plugin-transform-runtime/6.7.5/babel-plugin-transform-runtime

├── babel-preset-es2015 -> .npminstall/babel-preset-es2015/6.6.0/babel-preset-es2015
├── babel-preset-es2015-loose -> .npminstall/babel-preset-es2015-loose/7.0.0/babel-preset-es2015-loose
├── babel-preset-stage-1 -> .npminstall/babel-preset-stage-1/6.5.0/babel-preset-stage-1
├── babel-runtime -> .npminstall/babel-runtime/6.6.1/babel-runtime
├── cheerio -> .npminstall/cheerio/0.20.0/cheerio
├── debug -> .npminstall/debug/2.2.0/debug
├── request -> .npminstall/request/2.72.0/request
└── superagent -> .npminstall/superagent/1.8.3/superagent

12 directories, 0 files
```

after `npm i babel-eslint` at `~/.atom/packages/linter-eslint/` :

```
Error: Cannot find module 'eslint-config-airbnb/base' Referenced from: /Users/willin/Documents/w2fs/atl/.eslintrc.json
Error: Cannot find module 'eslint-config-airbnb/base'
Referenced from: /Users/willin/Documents/w2fs/atl/.eslintrc.json
    at Object.ModuleResolver.resolve (/Users/willin/.atom/packages/linter-eslint/node_modules/eslint/lib/util/module-resolver.js:77:19)
    at resolve (/Users/willin/.atom/packages/linter-eslint/node_modules/eslint/lib/config/config-file.js:475:33)
    at load (/Users/willin/.atom/packages/linter-eslint/node_modules/eslint/lib/config/config-file.js:492:24)
    at /Users/willin/.atom/packages/linter-eslint/node_modules/eslint/lib/config/config-file.js:388:36
    at Array.reduceRight (native)
    at applyExtends (/Users/willin/.atom/packages/linter-eslint/node_modules/eslint/lib/config/config-file.js:365:28)
    at Object.load (/Users/willin/.atom/packages/linter-eslint/node_modules/eslint/lib/config/config-file.js:527:22)
    at loadConfig (/Users/willin/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:67:33)
    at new Config (/Users/willin/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:200:38)
    at CLIEngine.executeOnText (/Users/willin/.atom/packages/linter-eslint/node_modules/eslint/lib/cli-engine.js:655:28)
```

**Comment by Arcanemagus** [comment]
First of all you should uninstall `babel-eslint` from `~/.atom/packages/linter-eslint`, that is _not_ the way to solve any of this!

It looks like from your `tree node_modules` that you have `babel-eslint` installed there... but you don't actually have `eslint` itself installed. Try adding that to your project's `package.json` and `linter-eslint` should load that version.

**Comment by canda** [comment]
Are you talking about node_modules in atom packages, or in your project?
I'm getting the same error but I don't have the babel-tslint module in either of that node_modules folder

```
Error: Cannot find module 'babel-eslint'
    at Object.ModuleResolver.resolve (/home/canda/.atom/packages/linter-eslint/node_modules/eslint/lib/util/module-resolver.js:75:19)
    at Object.load (/home/canda/.atom/packages/linter-eslint/node_modules/eslint/lib/config/config-file.js:513:42)
    at loadConfig (/home/canda/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:64:33)
    at new Config (/home/canda/.atom/packages/linter-eslint/node_modules/eslint/lib/config.js:197:38)
    at CLIEngine.executeOnText (/home/canda/.atom/packages/linter-eslint/node_modules/eslint/lib/cli-engine.js:744:28)
    at Object.cli.execute (/home/canda/.atom/packages/linter-eslint/node_modules/eslint/lib/cli.js:181:36)
    at lintJob (/home/canda/.atom/packages/linter-eslint/lib/worker.js:30:10)
    at /home/canda/.atom/packages/linter-eslint/lib/worker.js:66:20
    at Emitter.emit (/home/canda/.atom/packages/linter-eslint/node_modules/process-communication/node_modules/sb-event-kit/lib/emitter.js:70:19)
    at /home/canda/.atom/packages/linter-eslint/node_modules/process-communication/node_modules/sb-communication/lib/main.js:22:23
```

**Comment by Arcanemagus** [comment]
@canda you should have `eslint` and `babel-eslint` listed as dependencies of your project.

**Comment by framerate** [comment]
This is definitely caused by forgetting 'eslint' in package.json. Just verified on mine. Make sure BOTH are in package.json and the error should go away

**Comment by Arcanemagus** [comment]
Marking as closed due to lack of response (and the fact that the solution is almost certainly as @framerate describes above).

**Comment by tuanh118** [comment]
@Arcanemagus @framerate Should eslint and babel-eslint only appear in devDependencies instead of dependencies? My atom works fine when I put them under dependencies, but complains if I move them to devDependencies.

**Comment by IanVS** [comment]
@tuanh118 they should usually be devDependencies of your project.  Can you open a new issue with all relevant details so we can help troubleshoot?

**Comment by tuanh118** [comment]
@IanVS Thank you so much for quick response! It's working fine now only after I restarted my laptop (restarting atom alone didn't work). Not sure why I had to do that though.
Here's my config in case it helps.
Ubuntu 16.04.1 LTS
Atom    : 1.9.2
Electron: 0.37.8
Chrome  : 49.0.2623.75
Node    : 5.10.0

**Comment by TonyNikolov** [comment]
Just install babel-eslint -g, fixed it for me.

**Comment by eTorAken** [comment]
If you install it globally, don't forget to activate "Use global ESLint installation" option in `linter-eslint` settings:
![Don't forget this setting](https://i.gyazo.com/28d47f50e140487036e8e79318bd0fc5.png)

**Comment by yataras** [comment]
I've fixed it with running `npm install babel-eslint` in `~/.atom/packages/linter-eslint/`

**Comment by Arcanemagus** [comment]
@yataras As has been stated many times in this thread, that is _not_ the way to fix this. If you run the debug command on a file you were having problems on part of the information generated is the location of the ESLint instance in use.

Since installing it in this package's fallback copy "worked", it sounds like you haven't installed ESLint in your project which would be the ideal way to fix this. If you are having problems figuring out getting that to work feel free to file a new issue including the debug output.

**Comment by shadowsyntax** [comment]
@yataras Install babel-eslint as a devDependency and change the setting of linter-eslint on Atom not to use global ESLint installation.

**Comment by metakermit** [comment]
I'm getting these errors even if I have no eslint or babel-eslint packages installed locally in a project (and no *.eslintrc*). I have the option "Disable when no ESLint config is found (in package.json or .eslintrc)" set to true in my linter-eslint preferences.

Specifically, this is the repo I have checked out – https://github.com/airbnb/react-sketchapp/tree/master/examples/basic-setup

I'd really expect to not see any errors or other ESLint activity in this case. I now have to disable linter-eslint manually to stop random underlining of projects where I have no ESLint set up.

**Comment by IanVS** [comment]
@metakermit can you please post the output of the `Linter Eslint: Debug` command (from the command palette) when you have one of the files in that project open?

**Comment by skylize** [comment]
@metakermit 
> I'm getting these errors even if I have no eslint or babel-eslint packages installed locally in a project (and no .eslintrc). I have the option "Disable when no ESLint config is found (in package.json or .eslintrc)" set to true in my linter-eslint preferences.

The search for the config is not dependent on what Atom considers to be a project. We start at the linted file's location, and traverse upward through parent folders in search of a config. So if your project is inside a folder that has a config, that config will be found and used as local.

If you have `Disable when no ESLint config` setting enabled, and still getting error from a project with no config, that means you have your project inside another folder that has a config, tricking `linter-eslint` into thinking that the ancestor directory is your project.

If this does not solve your problem, please open a new and separate Issue, and include the output from Linter Eslint: Debug.

**Comment by metakermit** [comment]
@skylize ah, ok. That explains it, because the example folder I was in has another package.json in the parent folder. After I ran `npm install` in it, ESLint started working properly even if I open these example subfolders. Thanks for the clarification.
